tirocks/src/metadata.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    mem::MaybeUninit,
+    ops::Index,
+    str::{self, Utf8Error},
+};
+
+use tirocks_sys::{
+    crocksdb_livefiles_t, r, rocksdb_ColumnFamilyMetaData, rocksdb_LevelMetaData,
+    rocksdb_LiveFileMetaData, rocksdb_SizeApproximationOptions, rocksdb_SstFileMetaData, s,
+};
+
+use crate::RawDb;
+
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct SizeApproximationOptions(rocksdb_SizeApproximationOptions);
+
+impl SizeApproximationOptions {
+    /// Defines whether the returned size should include the recently written
+    /// data in the mem-tables. If set to false, include_files must be true.
+    #[inline]
+    pub fn set_include_memtables(&mut self, include: bool) -> &mut Self {
```

Comment:
can we extend the `simple_access!` and let these get generated

Reply:
`simple_access` is to make it easy to delegate functions to C ABI.

Reply:
yeah, I know. But these getters and setters can be generated by macro too. Maybe add a new macro?

Reply:
Personally I prefer avoiding using macros as it will hide the details, confuse the IDE and mess the error messages. The point of `simple_access` is because there are a lot of setters for options and C ABI is very long and easy to be written wrong, for example `crocksdb_set_min_xxxx` and `crocksdb_set_max_xxx`, without any notice of compilation errors.